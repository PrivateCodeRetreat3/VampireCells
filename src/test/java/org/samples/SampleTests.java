package org.samples;


import com.spun.util.ArrayUtils;
import com.spun.util.NumberUtils;
import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.lambda.utils.Grid;

import java.util.ArrayList;
import java.util.List;

public class SampleTests
{
//  @Test
//  public void testNormalCellDiesWhenLonely()
//  {
//    final NormalCell[] possible = new NormalCell[] {new NormalCell(1), new NormalCell(0), new NormalCell(99), new VampireCell() };
//    CombinationApprovals.verifyAllCombinations(this::getNext, possible, possible, possible, possible, possible, possible, possible, possible, possible);
//  }

  @Test
  public void testNeighborScenarios()
  {
    final NormalCell[] possible = new NormalCell[] {new NormalCell(1), new NormalCell(0), new NormalCell(99), new VampireCell() };
    CombinationApprovals.verifyAllCombinations(this::getNext, possible, new Integer[]{0, 1, 2, 3, 4}, new Integer[]{0, 1, 2, 3});
  }
  private String getNext(NormalCell center, NormalCell topLeft, NormalCell top, NormalCell topRight, NormalCell left,
                         NormalCell right, NormalCell bottomLeft, NormalCell bottom, NormalCell bottomRight) {
    final NormalCell next = center.next(List.of(topLeft, topRight, left, top, right, bottomLeft, bottom, bottomRight));
    return next.toString();
  }
  private String getNext(NormalCell center, Integer aliveNeighborCount, Integer vampireNeihborCount) {
    int deadNeighborCount = 8 - aliveNeighborCount - vampireNeihborCount;
    List<NormalCell> neighbors = new ArrayList<>();
    for (int i = 0; i < aliveNeighborCount; i++) {
      neighbors.add(new NormalCell(1));
    }
    for (int i = 0; i < vampireNeihborCount; i++) {
      neighbors.add(new VampireCell());
    }
    for (int i = 0; i < deadNeighborCount; i++) {
      neighbors.add(new NormalCell(0));
    }
    neighbors = List.of(NumberUtils.getShuffled(neighbors.toArray(new NormalCell[8]), 8));
    return center.next(neighbors).toString();
  }

}
