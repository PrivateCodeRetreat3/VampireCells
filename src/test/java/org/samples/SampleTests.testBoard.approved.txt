Initial:
__ __ __ __ __ 
__ 97 __ __ __ 
__ 96 __ __ __ 
__ 95 __ V  __ 
__ __ __ __ __ 


Frame #1:
__ __ __ __ __ 
__ __ __ __ __ 
01 97 01 __ __ 
__ __ __ V  __ 
__ __ __ __ __ 


Frame #2:
__ __ __ __ __ 
__ 01 __ __ __ 
__ 98 V  __ __ 
__ 01 __ V  __ 
__ __ __ __ __ 


Frame #3:
__ __ __ __ __ 
__ V  __ __ __ 
01 99 V  __ __ 
__ V  __ V  __ 
__ __ __ __ __ 


Frame #4:
__ __ __ __ __ 
__ V  __ __ __ 
V  V  __ __ __ 
__ V  __ V  __ 
__ __ __ __ __ 


Frame #5:
__ __ __ __ __ 
__ __ __ __ __ 
__ __ __ __ __ 
__ __ __ V  __ 
__ __ __ __ __ 


Frame #6:
__ __ __ __ __ 
__ __ __ __ __ 
__ __ __ __ __ 
__ __ __ V  __ 
__ __ __ __ __ 
